examples/gitserver/README.md

```diff
@@ -1,16 +1,207 @@
+   ```sh
+   $ git config --global credential.http://gitserver-myproject.infra.openshift.com.helper \
+         '!f() { echo "username=$(oc whoami)"; echo "password=$(oc whoami -t)"; }; f'
+   ```
+
+   **NOTE:** the config key is `credential.[git server URL].helper`
+
+3. Push a repository to your git server
+
+   In an existing repository, add a remote that points to the git server and push to it
+   
+   ```
+   # clone a public repository
+   $ git clone https://github.com/openshift/ruby-hello-world.git
+
+   # add a remote for your git server
+   $ cd ruby-hello-world
+   $ git remote add openshift http://git-myproject.infra.openshift.com/ruby-hello-world.git  
+
+   # push the code to the git server
+   $ git push openshift master
+   ```
+
+   **NOTE:** the ruby-hello-world.git repository does not exist before running these commands. 
+   By pushing to it, you are creating it in the Git Server.
+
+   On push the git server will invoke new-app on the code and create artifacts for it in 
+   OpenShift.
+
+
+### Secure the Git Server
+
+1. If your cluster's router has been setup with a default certificate for edge termination,
+   just modify the route to use edge termination:
+
+   ```sh
+   $ oc edit route git
+   ```
+
+   Add tls -> termination -> edge to the route specification:
+
+   ```yaml
+   apiVersion: v1
+   kind: Route
+   metadata:
+     name: gitserver
+   spec:
+     host: gitserver-myproject.infra.openshift.com
+     tls:
+       termination: edge
+     to:
+       kind: Service
+       name: gitserver
+   ```
+
+2. If using a private certificate authority, configure your git client to use the private ca.crt file:
+
+   ```sh
+   $ git config --global http.https://gitserver-myproject.infra.openshift.com.sslCAInfo /path/to/ca.crt
+   ```
+
+   where the key is http.[git server URL].sslCAInfo
+
+3. Disable anonymous cloning. By default the gitserver will allow anonymous cloning to make it easier to 
+   run builds without having to specify a secret. You can disable this by setting the `ALLOW_ANON_GIT_PULL`
+   environment variable to `no`.
+
+   ```sh
+   $ oc set env dc/git ALLOW_ANON_GIT_PULL=no
+   ```
+
+   **NOTE:** changing environment variables on the git server will cause a redeploy of the git server. If using
+   the default ephemeral storage for it, all repositories that have been pushed previously will be wiped out.
+   They will need to be pushed to the server again to restore.
+
+Authentication
+--------------
+
+By default, the gitserver will authenticate using OpenShift user or service account credentials. For a user,
+the credentials are the user name, and the user's token (from `oc whoami -t`). For a service account, it can
+be one of the 2 tokens created with the service account and stored in the service account secrets. These can 
+be obtained by looking at the secrets that correspond to the service account (`oc get secrets`) and displaying
+the token in one of them: `oc describe secret secret-name`.
```

Comment:
what do i do w/ this token?  how do i tell the gitserver whether to use a user account or a service account, and which service account?